### PR TITLE
TST: follow-up to Test nested pandas array #24993

### PR DESCRIPTION
**Pull request by simonjayhawkins**
follow-up to #24993

@jreback : following discussion with @jorisvandenbossche at europandas it was deemed undesirable to apply markers directly to tests in the base class since these tests can be used by downstream projects.

The main goal of this PR is to remove the duplication added in #24993. With the constraint noted above this poses a slight problem regarding parameterised base tests. The parameterisation needs to be duplicated! To reduce this slightly some parameterisation has been replaced with fixtures.

@TomAugspurger : the additions to `test_numpy.py` are moreless copied from `numpy_/test_numpy_nested.py` with a few additional comments on the failure cases and the numpy version skip scoped to only the tests using the `data_missing` fixture.

### CODE REVIEW

File: pandas/tests/extension/test_numpy.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+import pandas as pd
+from pandas import compat
+from pandas.core.arrays.numpy_ import PandasArray, PandasDtype
+import pandas.util.testing as tm
+
+from . import base
+
+
+NP_VERSION_INFO = tuple(int(x) for x in np.__version__.split('.'))
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/tests/extension/test_numpy.py]: i see that LooseVersion is used throughout the codebase. I'll change to use that instead.

**Review by jreback — CHANGES_REQUESTED** [review]
@simonjayhawkins looks good. some very minor comments. ping on green.

**Comment by jreback** [comment]
also pls merge master

**Review by jreback — APPROVED** [review]

**Comment by gfyoung** [comment]
Thanks @simonjayhawkins !

**Pull request merged**